server/tests/test_permissions.py

```diff
@@ -0,0 +1,167 @@
+import time
```

Comment:
It's good to test stuff without the mocks, but I think things will move quicker and smoother if you follow the pattern of testing authorization in two completely separate parts, 

1. Adding `patch_rules()` in ordinary request-based tests, making sure to test the 200 and 400 cases. This is not just a quick and dirty way to get tests passing (which you are doing), but also a way to ensure requests are rejected when the rule doesn't pass (which you aren't doing). It also insulates those tests from rule behaviour changes. 
2. `test_permissions.py` need only worry about testing rule implementations themselves, by running assertions against the non-mocked `test_rule` directly with concrete users, branches and indicators. It doesn't need to worry about views if we cover those in 1)